snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/components/SnomedRelationshipApiTest.java

```diff
@@ -152,13 +153,27 @@ public void createDuplicateRelationship() {
 
 	@Test
 	public void createRelationshipInferred() {
+		
 		Json requestBody = createRelationshipRequestBody(Concepts.ROOT_CONCEPT, Concepts.PART_OF, Concepts.NAMESPACE_ROOT, Concepts.INFERRED_RELATIONSHIP)
 				.with("commitComment", "Created new relationship with inferred characteristic type");
 
 		String relationshipId = assertCreated(createComponent(branchPath, SnomedComponentType.RELATIONSHIP, requestBody));
 
-		getComponent(branchPath, SnomedComponentType.RELATIONSHIP, relationshipId).statusCode(200)
-		.body("characteristicTypeId", equalTo(Concepts.INFERRED_RELATIONSHIP));
+		SnomedRelationship relationship = getComponent(branchPath, SnomedComponentType.RELATIONSHIP, relationshipId, "source()", "type()", "destination()", "characteristicType()", "modifier()")
+			.statusCode(200)
+			.extract().as(SnomedRelationship.class);
+		
+		assertNotNull(relationship.getCharacteristicType());
+		assertEquals(Concepts.INFERRED_RELATIONSHIP, relationship.getCharacteristicType().getId());
+		assertNotNull(relationship.getSource());
+		assertEquals(Concepts.ROOT_CONCEPT, relationship.getSource().getId());
+		assertNotNull(relationship.getType());
+		assertEquals(Concepts.PART_OF, relationship.getType().getId());
+		assertNotNull(relationship.getDestination());
+		assertEquals(Concepts.NAMESPACE_ROOT, relationship.getDestination().getId());
```

Comment:
For concept-valued properties, a minimal `SnomedConcept` instance is always present, even without expanding (the example is for `setDestinationId` but other setters are also put together this way): https://github.com/b2ihealthcare/snow-owl/blob/2db98207d4ab37615c3fcabfe25dfddd22897be6/snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/SnomedRelationship.java#L299-L302

 I'd recommend adding a non-null check for eg. the `active` status or `moduleId` property for each one to be sure that they were fully populated with information.

Reply:
I would remove the non-null checks completely, as they don't add that much to the test itself.

Reply:
Old habits of getPt() throwing NPE tricked me 😊, removed the null checks 🎉